src/coreclr/gcinfo/gcinfoencoder.cpp

```diff
@@ -933,7 +937,34 @@ bool GcInfoEncoder::IsAlwaysScratch(GcSlotDesc &slotDesc)
         _ASSERTE(regNum >= 0 && regNum <= 14);
         _ASSERTE(regNum != 13);  // sp
 
-        return ((regNum <= 3) || (regNum >= 12)); // R12 and R14/LR are both scratch registers
+        return ((regNum <= 3) || (regNum >= 12)) // R12 is volatile and SP/LR can't contain objects around calls
```

Comment:
ARM/ARM64 has more changes than x64 here because some of this is just to get the filtering to the same level as on x64.   (looks like we had some NYIs)

Otherwise, it would be just opt-in for X0/X1